src/app/table/table.component.ts

```diff
@@ -129,7 +139,17 @@ export class GpfTableComponent implements OnChanges, AfterViewChecked {
   }
 
   public getVisibleData(): Array<any> {
-    this.tableWidth = this.tableViewChild?.nativeElement.offsetWidth;
+    this.tableWidth = '0';
+    if (this.columnsChildren !== undefined && this.columnsChildren !== null) {
+      this.columnsChildren.forEach(column => {
+        this.tableWidth = String(parseFloat(this.tableWidth) + parseFloat(column.columnWidth));
+      });
+      this.tableWidth += 'px';
+      if (parseFloat(this.tableWidth) <= this.tableViewChild?.nativeElement.offsetWidth) {
+        this.tableWidth = this.tableViewChild?.nativeElement.offsetWidth;
+      }
```

Comment:
Perhaps the final conversion of `tableWidth` to a string value should be done after this `if` to avoid yet another conversion to float